### PR TITLE
ci(CI): :green_heart: Added azure-pipelines configuration

### PR DESCRIPTION
**Pull request by SarantopoulosKon**
Azure-pipelines yaml for automating releases with the use of semantic
release package. Semantic release decides whether to release a new
version, auto generates CHANGELOG.md and publishes the package.

### CODE REVIEW

File: azure-pipelines.yaml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  - task: NodeTool@0
+    inputs:
+      versionSpec: '12.x'
+  - script: |
+      yarn
+    displayName: 'Install dependencies'
+  - script: |
+      yarn build
+    displayName: 'Build'
+  - script: |
```
Comment (Shelnutt2) [inline azure-pipelines.yaml:154]: The scripts here need the `condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags'))` protection

File: azure-pipelines.yaml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    vmImage: $(imageName)
+
+  steps:
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: '$(python.version)'
+      architecture: 'x64'
+
+  - script: brew install node@12
+    displayName: 'Install NodeJS'
```
Comment (Shelnutt2) [inline azure-pipelines.yaml:49]: brew commands should be limited only to osx

**Review by Shelnutt2 — CHANGES_REQUESTED** [review]

File: azure-pipelines.yaml (hunk 2 above)
Comment (SarantopoulosKon) [reply azure-pipelines.yaml:49]: that's true i've added a condition `condition: eq(variables['imageName'], 'macOS-10.14')` to install with brew and for ubuntu i added another script

File: azure-pipelines.yaml (hunk 1 above)
Comment (SarantopoulosKon) [reply azure-pipelines.yaml:154]: Added the condition for both build and release

File: azure-pipelines.yaml (hunk 2 above)
Comment (Shelnutt2) [reply azure-pipelines.yaml:49]: Use `Agent.OS` to prevent this from breaking when we one day upgrade to `10.15`

File: azure-pipelines.yaml [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+- job: Deploy_js
+  pool:
+    vmImage: 'ubuntu-16.04'
+
+  steps:
+  - task: NodeTool@0
+    inputs:
+      versionSpec: '12.x'
+  - script: |
+      yarn
```
Comment (Shelnutt2) [inline azure-pipelines.yaml:142]: Does running just yarn install the dependencies?

File: azure-pipelines.yaml (hunk 1 above)
Comment (Shelnutt2) [reply azure-pipelines.yaml:154]: Perfect thanks!

**Review by Shelnutt2 — COMMENTED** [review]
One comment, then go ahead and squash all your commits

**Pull request merged**